create_graphics.py

```diff
@@ -530,6 +535,198 @@ def parallel_skewt(cla, fhr, ds, site, workdir):
+            os.path.exists(grib_path) else False
+        return grib_path, old_enough
+
+    # Generate a list of files to be joined.
+    file_list = [os.path.join(*path).format(FCST_TIME=fhr) for path in
+                 zip(cla.data_root, cla.file_tmpl)]
+    for file_path in file_list:
+        if not os.path.exists(file_path) \
+            or not utils.old_enough(cla.data_age, file_path):
+            return file_path, False
+
+    print(f'Combining input files: ')
+    for fn in file_list:
+        print(f'  {fn}')
+
+    combined_fp = os.path.join(cla.output_path,
+                               COMBINED_FN.format(fhr=fhr))
+    tmp_fp = os.path.join(cla.output_path,
+                          TMP_FN.format(fhr=fhr))
+
+    cmd = f'cat {" ".join(file_list)} > {tmp_fp}'
+    output = subprocess.run(cmd,
+                            capture_output=True,
+                            check=True,
+                            shell=True,
+                            )
+    if output.returncode != 0:
+        msg = f'{cmd} returned exit status: {output.returncode}!'
+        raise OSError(msg)
+
+    # Gather all grib2 entries from combined file
+    cmd = f'wgrib2 {tmp_fp} -submsg 1'
+    output = subprocess.run(cmd,
+                            capture_output=True,
+                            check=True,
+                            shell=True,
+                            )
+    wgrib2_list = output.stdout.decode("utf-8").split('\n')
+
+    # Create a unique list of grib fields.
+    uniq_list = uniq_wgrib2_list(wgrib2_list)
+
+    # Remove duplicate grib2 entries in grib file
+    cmd = f'wgrib2 -i {tmp_fp} -GRIB {combined_fp}'
+    input_arg = '\n'.join(uniq_list).encode("utf-8")
+
+    output = subprocess.run(cmd,
+                            capture_output=True,
+                            check=True,
+                            input=input_arg,
+                            shell=True,
+                            )
+    if output.returncode != 0:
+        msg = f'{cmd} returned exit status: {output.returncode}'
+        raise OSError(msg)
+    os.remove(f'{tmp_fp}')
+
+    return f'{combined_fp}', True
+
+def remove_accumulated_images(cla):
+
+    ''' Searches for all images that correspond with specs that have the
+    accumulate entry set to True and removes them from the list of images to
+    create. '''
+
+    for variable, levels in cla.images[1].items():
+        for level in levels:
+            spec = cla.specs.get(variable, {}).get(level)
+            if not spec:
+                msg = f'graphics: {variable} {level}'
+                raise errors.NoGraphicsDefinitionForVariable(msg)
+            accumulate = spec.get('accumulate', False)
+
+            if accumulate:
+                print(f'Will not plot {variable}:{level}')
+                cla.images[1][variable].remove(level)
+                if not cla.images[1][variable]:
+                    del cla.images[1][variable]
+
```

Comment:
I feel slightly icky about this implementation where I'm changing input arguments explicitly. I may want to clean that up more in a subsequent PR. I think it works okay for now. Happy to hear feedback.